haystack/components/converters/openapi_functions.py

```
@@ -0,0 +1,177 @@
+    - Each function must have a unique operationId
+    - Each function must have a description
+    - Each function must have a requestBody or parameters or both
+    - Each function must have a schema for the requestBody and/or parameters
+
+
+    See https://github.com/OAI/OpenAPI-Specification for more details on OpenAPI specification.
+    See https://platform.openai.com/docs/guides/function-calling for more details on OpenAI function calling.
+    """
+
+    def __init__(self):
+        """
+        Initializes the OpenAPIServiceToFunctions instance
+        """
+        openapi_imports.check()
+
+    @component.output_types(documents=List[Document])
+    def run(self, sources: List[Union[str, Path, ByteStream]], system_messages: List[str]) -> Dict[str, Any]:
+        """
+        Processes OpenAPI specification URLs or files to extract functions that can be invoked via OpenAI function
+        calling mechanism. Each source is paired with a system message in one-to-one correspondence. The system message
+        is used to assist LLM in the response generation.
+
+        :param sources: A list of OpenAPI specification sources, which can be URLs, file paths, or ByteStream objects.
+        :type sources: List[Union[str, Path, ByteStream]]
+        :param system_messages: A list of system messages corresponding to each source.
+        :type system_messages: List[str]
+        :return: A dictionary with a key 'documents' containing a list of Document objects. Each Document object
+                 encapsulates a function definition and relevant metadata.
+        :rtype: Dict[str, Any]
+        :raises RuntimeError: If the OpenAPI specification cannot be downloaded or processed.
+        :raises ValueError: If the source type is not recognized or no functions are found in the OpenAPI specification.
+        """
+        documents: List[Document] = []
+        for source, system_message in zip(sources, system_messages):
+            try:
```

Can we make this `try..except` a bit smaller? It could hide legitimate errors this way.